Only observe scrollview while pullToRefreshView is a subview.

chapados opened the pull request:
KVO for scrollview properties was previously started in -init and removed in -dealloc.
However, the timing of dealloc is not determinant, since it may be called on another
thread.  In cases where both the pullToRefresh and infiniteScrollingView are
present, it is possible to reach a condition where the scrollview (e.g. tableview)
is deallocated before KVO properties are removed. Since the
pullToRefresh views are stored in associated objects on the scrollview, it is also
not clear exactly when the associated objects will be released.
By implementing -[UIView willMoveToSuperview:] and managing KVO properties there, we ensure
that KVO is directly coupled to state of the scrollview's hierarchy. Note that we
still include a call to -stopObservingScrollview in -dealloc.  This guards against
cases where UIView notifications may not be sent and is otherwise a no-op.

chapados committed:
Only observe scrollview while pullToRefreshView is a subview.

KVO for scrollview properties was previously started in -init and removed in -dealloc.
However, the timing of dealloc is not determinant, since it may be called on another
thread.  In cases where both the pullToRefresh and infiniteScrollingView are
present, it is possible to reach a condition where the scrollview (e.g. tableview)
is deallocated before KVO properties are removed. Since the
pullToRefresh views are stored in associated objects on the scrollview, it is also
not clear exactly when the associated objects will be released.
By implementing -[UIView willMoveToSuperview:] and managing KVO properties there, we ensure
that KVO is directly coupled to state of the scrollview's hierarchy. Note that we
still include a call to -stopObservingScrollview in -dealloc.  This guards against
cases where UIView notifications may not be sent and is otherwise a no-op.

samvermette commented:
@chapados excellent call, thank you. Slight detail: as you can see I'm not using ivars in my project (all properties instead), so the way you're declaring `_observingScrollView` is kind of disconnected with the rest of the project. It's my understanding that contributing to open-source projects typically involves adapting to the original project's coding style :)

If you have a sec to fix this I would really appreciate, otherwise I'll fix it myself when I get the chance :)

chapados committed:
Use an observingScrollView property instead of an ivar.

This makes the scrollView KVO patch conform to the existing
style of SVPullToRefresh.

chapados commented:
@samvermette Done! Sorry about that. Let me know if this additional commit is fine, or if you'd rather have me rebase these together and make a new pull request.  Thanks for the fast response!

samvermette commented:
@chapados someone came up with an [issue](https://github.com/samvermette/SVPullToRefresh/issues/37) regarding this change, says moving `[self startObservingScrollView]` to `initWithScrollView` fixes it. Any clue?
